internal/elasticsearch/ingest/datastream.go

```diff
@@ -28,9 +28,9 @@ var (
 )
 
 type Rule struct {
-	TargetDataset interface{} `yaml:"target_dataset"`
-	If            string      `yaml:"if"`
-	Namespace     interface{} `yaml:"namespace"`
+	TargetDataset *interface{} `yaml:"target_dataset"`
+	If            *string      `yaml:"if"`
+	Namespace     *interface{} `yaml:"namespace"`
```

Comment:
An `interface{}` can be checked directly against nil, we wouldn't need a pointer to it.
```suggestion
	TargetDataset interface{} `yaml:"target_dataset"`
	If            *string      `yaml:"if"`
	Namespace     interface{} `yaml:"namespace"`
```

Or would it be possible to use omitempty instead?
```suggestion
	TargetDataset interface{} `yaml:"target_dataset,omitempty"`
	If            string      `yaml:"if,omitempty"`
	Namespace     interface{} `yaml:"namespace,omitempty"`
```

Reply:
I guess it's about when `convertValue` is called and when field is empty it doesn't fall into any of the switch cases. So now since I added a check in `convertValue` for `if originalValue == nil`, it's all good :)